dashboard/test/models/census/ap_school_code_test.rb

```diff
@@ -40,4 +40,45 @@ class Census::ApSchoolCodeTest < ActiveSupport::TestCase
     school_code = Census::ApSchoolCode.normalize_school_code("0123")
     assert_equal "000123", school_code
   end
+
+  test "AP school code year is missing fails" do
+    school_code = build :ap_school_code, :without_school_year
+    refute school_code.valid?
+  end
+
+  test "AP school code year is invalid fails" do
+    school_code = build :ap_school_code, :with_invalid_school_year
+    refute school_code.valid?
+  end
+
+  test "Composite key relationship between AP school code and AP CS offering" do
```

Comment:
@sureshc added another test that mirrors some of the scenarios you whiteboarded yesterday. I have three assertions here -- should I split into different tests?